full-service/src/service/transaction.rs

```diff
@@ -213,8 +226,20 @@ where
 
         let tx_proposal = builder.build()?;
 
-        // FIXME: WS-32 - Might be nice to have a tx_proposal table so that you don't
-        // have to write these out to local files.
+        if let Some(should_log) = log_tx_proposal {
```

Comment:
Instead of nesting you should be able to do something like `if log_tx_proposal.unwrap_or_default(false) {`